README.md

```diff
@@ -1,4 +1,253 @@
+
+**App** is used to resolve the permission name and model.
+
+**Action** is an arbitrary identifier that is used to resolve the permission name.
+
+**Is Global** whether the permission is global or per-object (see "Global Permission" section below)
+
+**Evaluators**
+
+The next columns define permission "evaluators".
+
+Built-in evaluators are:
+
+* `all` - user has permission for all objects. Will raise an error  if an object is not passed to `has_perm()`
+* `yes` - user has permission globally. Will raise an error if an object is passed to `has_perm()`.
+* (empty cell) -- user does not have permission (global or per-object) 
+
+
+### Global Permissions
+
+Unlike vanilla django permissions, by default `cvs_permissions` imposes a hard
+distinction between global and per-object permissions.
+
+* If you pass an object in a permission check against a permission with
+    `Is Global==yes` in the CSV file then a `ValueError` will be raised.
+* If you *don't* pass an object to a permission check against a permission with
+  `Is Global==no` in the CSV file then a `ValueError` will be raised.
+
+The `CSVPermissionsBackend` provides an `is_global_perm()` method to query
+whether a permission is global or per-object:
+
+```python
+# example of querying whether a permission is global 
+print(
+    "foo-bar is a global permission"
+    if CSVPermissionBackend().is_global("foo-bar")
+    else "foo-bar is a per-object permission"
+)
+```
+
+### Custom Evaluators
+
+By default putting other than a built-in evaluator in a CSV permissions file
+will raise an error.
+
+You add your own permission evaluators by defining "evaluator resolver"
+functions which ingest a CSV cell value and returns a permission evaluator.
+If the resolver does not recognise something it should return `None` and the
+next resolver in the list will be called.
+
+```python
+# in settings.py
+CSV_PERMISSIONS_RESOLVE_EVALUATORS = (
+    # sanity checks
+    'csv_permissions.evaluators.resolve_validation_evaluator',
+    # custom validators (examples below)
+    'my_app.evaluators.resolve_evaluators',
+    # 'all'/'yes'/'' 
+    'csv_permissions.evaluators.resolve_all_evaluator',
+    'csv_permissions.evaluators.resolve_yes_evaluator',
+    'csv_permissions.evaluators.resolve_empty_evaluator',
+    # normally if nothing matches an exception will be thrown however it 
+    # can be more convenient (especially in early phases of development )
+    # to issue a warning during CSV parsing, and then throw a
+    # NotImplementedError() when the permission is evaluated
+    'csv_permissions.evaluators.resolve_fallback_not_implemented_evaluator',
+)
+
+# if you don't have any customisations you can point to a list/tuple
+# that is defined elsewhere; this is a basic set:
+#CSV_PERMISSIONS_RESOLVE_EVALUATORS = "csv_permissions.evaluators.default_resolve_evaluators"
```

Comment:
so you need to define this even if you have no customisations? or can you leave out `CSV_PERMISSIONS_RESOLVE_EVALUATORS` and it defaults automatically?

Reply:
Have clarified that this is the default setting if you don't set it